Packages/com.unity.inputsystem/InputSystem/Actions/InputActionAsset.cs

```diff
@@ -1014,60 +1014,125 @@ internal void MigrateJson(ref ReadFileJson parsedJson)
         {
             if (parsedJson.version >= JsonVersion.Version1)
                 return;
-            if ((parsedJson.maps?.Length ?? 0) > 0 && (parsedJson.version) < JsonVersion.Version1)
+
+            if (parsedJson.maps == null || parsedJson.maps.Length == 0)
+            {
+                parsedJson.version = JsonVersion.Version1;
+                return;
+            }
+
+            for (var mi = 0; mi < parsedJson.maps.Length; ++mi)
             {
-                for (var mi = 0; mi < parsedJson.maps.Length; ++mi)
+                var mapJson = parsedJson.maps[mi];
+                if (mapJson.actions == null || mapJson.actions.Length == 0)
+                    continue;
+
+                for (var ai = 0; ai < mapJson.actions.Length; ++ai)
```

Comment:
Would it be possible to add some inline comments to the different parts of this large for loop or split it up so that is clear what the intent is? It is quite difficult to review given that there is no motivation given for what we do here.